libs/community/langchain_community/tools/azure_ai_services/azure_translator/azureTranslate.py

```diff
@@ -0,0 +1,150 @@
+        "languages. Input must be text (str). Ensure to install the "
+        "azure-ai-translation-text package."
+    )
+
+    def __init__(
+        self,
+        *,
+        translate_key: Optional[str] = None,
+        translate_endpoint: Optional[str] = None,
+    ) -> None:
+        """
+        Initialize the AzureTranslateTool with the given API key and endpoint.
+        """
+        translate_key = translate_key or os.getenv("AZURE_OPENAI_TRANSLATE_API_KEY")
+        translate_endpoint = translate_endpoint or os.getenv(
+            "AZURE_OPENAI_TRANSLATE_ENDPOINT"
+        )
+
+        if not translate_key or not translate_endpoint:
+            raise ValueError("Missing API key or endpoint for Azure Translator API.")
+
+        super().__init__(
+            translate_key=translate_key, translate_endpoint=translate_endpoint
+        )
+
+        self.translate_client = TextTranslationClient(
+            endpoint=translate_endpoint, credential=AzureKeyCredential(translate_key)
+        )
+
+    def _translate_text(self, text: str, to_language: str) -> str:
+        """
+        Perform text translation using the Azure Translator API.
+
+        Args:
+            text (str): The text to be translated.
+            to_language (str): The target language to translate to.
+
+        Returns:
+            str: The translation result.
+        """
+        if not text:
+            logger.error("Input text for translation is empty.")
+            return None
+
+        body = [{"Text": text}]
+        try:
+            response = self.translate_client.translate(
+                body=body, to_language=[to_language]
+            )
+            if response:
+                logger.warning(
+                    f"Translation successful: {response[0].translations[0].text}"
+                )  # Use WARNING level for successful operations
+                return response[0].translations[0].text
+            else:
+                logger.error("Translation failed with an empty response")
+                return None
+        except Exception as e:
+            logger.error(f"Translation failed: {e}")
+            raise
+
+    def _run(
+        self,
+        query: str,
+        to_language: str,
+        run_manager: Optional[CallbackManagerForToolRun] = None,
+    ) -> str:
+        """
+        Run the tool to perform translation.
+
+        Args:
+            query (str): The text to be translated.
+            to_language (str): The target language to translate to.
+            run_manager (Optional[CallbackManagerForToolRun]): A callback manager
+            for tracking the tool run.
+
+        Returns:
+            str: The translated text.
+        """
+        try:
+            return self._translate_text(query, to_language)
+        except Exception as e:
+            raise RuntimeError(f"Error while running AzureTranslateTool: {e}")
+
+    @classmethod
+    def from_env(cls):
```

Comment:
was this just added for temp testing? It seems we should read the env var values in the validate_environment.